src/cascadia/TerminalApp/TerminalPage.h

```diff
@@ -94,15 +94,15 @@ namespace winrt::TerminalApp::implementation
         void _HookupKeyBindings(TerminalApp::AppKeyBindings bindings) noexcept;
         void _RegisterActionCallbacks();
 
-        void _UpdateTitle(std::shared_ptr<Tab> tab);
-        void _UpdateTabIcon(std::shared_ptr<Tab> tab);
+        void _UpdateTitle(winrt::TerminalApp::Tab tab);
+        void _UpdateTabIcon(winrt::TerminalApp::Tab tab);
         void _UpdateTabView();
         void _UpdateTabWidthMode();
         void _DuplicateTabViewItem();
         void _RemoveTabViewItem(const Microsoft::UI::Xaml::Controls::TabViewItem& tabViewItem);
         void _RemoveTabViewItemByIndex(uint32_t tabIndex);
 
-        void _RegisterTerminalEvents(Microsoft::Terminal::TerminalControl::TermControl term, std::shared_ptr<Tab> hostingTab);
+        void _RegisterTerminalEvents(Microsoft::Terminal::TerminalControl::TermControl term, winrt::com_ptr<Tab> hostingTab);
```

Comment:
prefer `const winrt::com_ptr<Tab> &` unltil proven that you need to take an owning copy of the smart pointer